autometa/config/databases.py

```diff
@@ -66,230 +97,504 @@ class Databases:
 
         """
-        if not self.config.has_section('databases'):
-            self.config.add_section('databases')
-        for section in Databases.SECTIONS:
-            if self.config.has_option('databases', section):
-                continue
-            outdir = DEFAULT_CONFIG.get('databases', section)
-            self.config.set('databases', section, outdir)
+        any_missing = self.get_missing(section=section, validate=True)
+        if compare_checksums:
+            any_invalid = self.compare_checksums(section=section, validate=True)
+        else:
+            any_invalid = False
+        if not any_missing and not any_invalid:
+            return True
+        else:
+            return False
+
+    def get_remote_checksum(self, section, option):
+        """Get the checksum from provided `section` respective to `option` in `self.config`.
+
+        Parameters
+        ----------
+        section : str
+            section to retrieve for `checksums` section.
+            Choices include: 'ncbi' and 'markers'.
+        option : str
+            `option` in `checksums` section corresponding to the section checksum file.
+
+        Returns
+        -------
+        str
+            checksum of remote md5 file. e.g. 'hash filename\n'
+
+        Raises
+        -------
+        ConnectionError
+            Failed to connect to host for provided `option`.
+
+        """
+        if section not in {'ncbi','markers'}:
+            raise ValueError(f'"section" must be "ncbi" or "markers". Provided: {section}')
+        if section == 'ncbi':
+            host = self.config.get(section, 'host')
+            ftp_fullpath = self.config.get('checksums', option)
+            chksum_fpath = ftp_fullpath.split(host)[-1]
+            with FTP(host) as ftp, tempfile.TemporaryFile() as fp:
+                ftp.login()
+                result = ftp.retrbinary(f'RETR {chksum_fpath}', fp.write)
+                if not result.startswith('226 Transfer complete'):
+                    raise ConnectionError(f'{chksum_fpath} download failed')
+                ftp.quit()
+                fp.seek(0)
+                checksum = fp.read().decode()
+        elif section == 'markers':
+            url = self.config.get('checksums', option)
+            with requests.Session() as session:
+                resp = session.get(url)
+                if not resp.ok:
+                    raise ConnectionError(f'Failed to retrieve {url}')
+                checksum = resp.text
+        return checksum
 
     def format_nr(self):
-        """Format NCBI nr.gz database into diamond formatted database nr.dmnd.
+        """Construct a diamond formatted database (nr.dmnd) from `nr` option
+        in `ncbi` section in user config.
+
+        NOTE: The checksum 'nr.gz.md5' will only be generated if nr.dmnd
+        construction is successful. If the provided `nr` option in `ncbi` is 'nr.gz'
+        the database will be removed after successful database formatting.
```

Comment:
This is sort of confusing. Shouldn't we generate a md5 file of the `nr.dmnd` file?